---
title: "Supplementary_Figures"
output: html_document
date: "2024-09-29"
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

# loading libraries
```{r}
library(reshape2)
library(tidyverse)
library(ggplot2)
library(RColorBrewer)
library(gtable)
library(ggh4x)
library(broom)
library(vegan)
library(stringi)
library(gridExtra)
library(VennDiagram)
library(eulerr)
library(magrittr)
library(circlize)
library(ComplexHeatmap)
library(rsample)
library(randomForest)
library(ranger)
library(caret)
library(skimr)
library(RANN)
library(caretEnsemble)
library(combinat)
library(plotROC)
library(cowplot)
library(AmesHousing)
library(plotly)
library(segmented)
```

# Supplementary Figure 2d
```{r}
df_supp2d <- read.delim2(file = "/Users/chey2/Desktop/DOCK8_paper_codes/Input_files/Supp_2d.txt", header = TRUE, sep = "\t", stringsAsFactors = FALSE, check.names = FALSE)

df_supp2d %>% 
  mutate_at(2:5, as.numeric) -> df_supp2d

df_supp2d$site_specific <- factor(df_supp2d$site_specific, levels = c("Ra", "Mb", "Vf", "Hp", "Ac", "Ic", "Pc", "Ph"))

psupp2d <- df_supp2d %>% 
  melt() %>%
  ggplot(aes(x=site_specific, y=100-value, fill= factor(variable, levels = c("Standard", "Standard+SMGC", "Standard+DOCK8", "Standard+DOCK8+")))) +
  geom_boxplot(lwd = 0.05, outlier.size = 0.001, outlier.colour = "grey") +
  facet_wrap(~factor(Group, levels = c("Pre-Tx", "Post-Tx-onIS≤3M", "Post-Tx-onIS-6M", "Post-Tx-onIS≥12M", "Post-Tx-offIS-12M", "Post-Tx-offIS-24M", "HVs")), ncol=7) +
  theme_bw() +
  theme(axis.title.x=element_blank(), axis.text.x = element_text(size = 10), axis.title.y=element_text(size=12))+
  theme(axis.text.y = element_text(size = 10)) + 
  ylab("Percentage of classified reads (%)") +
  theme(panel.grid.minor=element_blank(), panel.grid.major=element_blank(), legend.title=element_text(size=12)) + theme(legend.text = element_text(size = 10)) + 
  scale_fill_manual(values = c("#7F58AF", "#64C5EB", "#FEB326", "#E84D8A"), name = "Database") 

psupp2d
```

# Supplementary Figure 3a
```{r}
df_supp3a <- read.delim("/Users/chey2/Desktop/DOCK8_paper_codes/Input_files/Supp_3_1_input.txt", header = TRUE, sep = "\t") # input
df_supp3a$Breadth <- factor(df_supp3a$Breadth, levels = c("35", "45", "55", "65", "75", "85"))
df_supp3a$Coverage <- factor(df_supp3a$Coverage, levels = c("2", "4", "6", "8", "10", "50", "100", "300"))

psupp3a_1 <- plot_ly(data = df_supp3a, x=~Breadth,y=~Coverage, z=~100*(1-mean_jaccard_new), type = "contour", colorscale = list(c(0, "rgba(0,0,255,0.95)"),
                          c(0.25, "rgba(0,255,255,0.95)"),
                          c(0.5, "rgba(255,255,0,0.95)"),
                          c(0.75, "rgba(255,0,0,0.95)"),
                          c(1, "rgba(255,255,255,0.95)")), autocontour = F, contours = list(
   coloring = 'heatmap'
  ), colorbar = list(title = "jaccard dissimilarity", len=0.4, thickness = 10)) 


psupp3a_2 <- plot_ly(data = df_supp3a, x=~Breadth,y=~Coverage, z=~100*(mean_scaled_eucli_new), type = "contour", colorscale = list(c(0, "rgba(0,0,255,0.95)"),
                          c(0.25, "rgba(0,255,255,0.95)"),
                          c(0.5, "rgba(255,255,0,0.95)"),
                          c(0.75, "rgba(255,0,0,0.95)"),
                          c(1, "rgba(255,255,255,0.95)")), autocontour = F, contours = list(
   coloring = 'heatmap'
  ), colorbar = list(title = "scaled euclidean dissimilarity", tickvals = c(1, 2, 3), ticktext = c("1", "2", "3"), len=0.4, thickness = 10)) 

psupp3a_1
psupp3a_2
```

# Supplementary Figure 5a, 5b
```{r}
# supplementary figure 5
df_supp5 <- read.delim("//Users/chey2/Desktop/DOCK8_paper_codes/Input_files/Supp_5_input.txt", header = TRUE, sep = "\t", colClasses = c("factor", "factor", "factor", "numeric")) # input

df_supp5$subject_id <- factor(df_supp5$subject_id, levels = c("Pt01", "Pt02", "Pt03", "Pt04", "Pt05", "Pt06", "Pt07", "Pt08", "Pt09", "Pt10", "Pt11", "Pt12", "Pt13", "Pt14", "Pt15", "Pt16", "Pt17", "Pt18", "Pt19", "Pt20", "Pt21", "Pt22", "Pt23", "Pt24", "Hv01", "Hv02", "Hv03", "Hv04", "Hv05", "Hv06", "DPt07", "DPt08", "DPt09", "DPt10", "DPt14", "DPt15", "DPt18"))
df_supp5$kingdom <- factor(df_supp5$kingdom, levels = c("Bacteria", "Fungi", "Phage", "Eukaryotic virus"))
df_supp5$Group <- factor(df_supp5$Group, levels = c("Pre-Tx", "Post-Tx-onIS-1M", "Post-Tx-onIS-2M", "Post-Tx-onIS-3M","Post-Tx-onIS-6M","Post-Tx-onIS>12M","Post-Tx-offIS-12M","Post-Tx-offIS-24M","HVs", "Donor"))

psupp5a <- df_supp5 %>% 
  filter(!Group=="Donor") %>%
  ggplot(aes(x=subject_id, y = mean_abu_norm_bs, fill = kingdom)) +
  geom_bar(stat = "identity", position = "fill", width = 0.8) +
  scale_y_continuous(labels = function(x) x*100) +
  facet_grid(~Group, scales = "free_x", space = "free_x") + 
  theme_bw() +
  theme(axis.text.x = element_text(angle = 90, size=8, hjust=1, vjust=0.3), axis.title.x=element_blank(), axis.text.y=element_text(size=8), axis.title.y=element_text(size=10)) +
  theme(panel.grid.minor=element_blank(), panel.grid.major=element_blank(), legend.title=element_text(size=10), legend.text=element_text(size=8)) + 
  ylab("Mean relative abundance (%)") +
  scale_fill_manual("Kingdom", values = c("#31a253", "#3073b9","#c793c2","#5d488b"))

psupp5a


df_supp5b <- read.delim("/Users/chey2/Desktop/DOCK8_paper_codes/Input_files/Supp_5b_input.txt", header = TRUE, sep = "\t", colClasses = c("factor", "factor", "factor", "numeric")) # input

df_supp5b$subject_id <- factor(df_supp5b$subject_id, levels = c("Pt01", "Pt02", "Pt03", "Pt04", "Pt05", "Pt06", "Pt07", "Pt08", "Pt09", "Pt10", "Pt11", "Pt12", "Pt13", "Pt14", "Pt15", "Pt16", "Pt17", "Pt18", "Pt19", "Pt20", "Pt21", "Pt22", "Pt23", "Pt24", "Hv01", "Hv02", "Hv03", "Hv04", "Hv05", "Hv06", "DPt07", "DPt08", "DPt09", "DPt10", "DPt14", "DPt15", "DPt18"))
df_supp5b$Group <- factor(df_supp5b$Group, levels = c("Pre-Tx", "Post-Tx-onIS-1M", "Post-Tx-onIS-2M", "Post-Tx-onIS-3M","Post-Tx-onIS-6M","Post-Tx-onIS>12M","Post-Tx-offIS-12M","Post-Tx-offIS-24M","HVs", "Donor"))
df_supp5b$Classification <- factor(df_supp5b$Classification, levels = c("Non-viruses", "Anelloviridae","Herpesviridae", "Papillomaviridae","Betapapillomavirus", "Gammapapillomavirus", "Polyomaviridae", "Alphapolyomavirus", "Deltapolyomavirus", "Poxviridae", "Molluscipoxvirus"))

colors <- c("#F0F0F0", "#E0E4CC", "#a8e6ce", "#BDD7E7","#3182BD", "#08519C", "#FEE6CE", "#FDD0A2", "#F16913", "#faefbe", "#fccf09")

psupp5b <- df_supp5b %>% 
  filter(!Group=="Donor") %>% 
  ggplot(aes(x = subject_id, y = mean_abu_nor, fill = Classification)) +
  geom_bar(stat = "identity", position = "fill", width = 0.8) + 
  facet_grid(~Group, scales = "free_x", space = "free_x") +
  scale_y_continuous(labels = function(x) x*100) +
  theme_bw() +
  theme(axis.text.x = element_text(angle = 90, size=8, hjust=1, vjust=0.3), axis.title.x=element_blank(), axis.text.y=element_text(size=8), axis.title.y=element_text(size=10)) +
  theme(panel.grid.minor=element_blank(), panel.grid.major=element_blank(), legend.title=element_text(size=10), legend.text=element_text(size=8)) +
  ylab("Mean relative abundance (%)") + 
  scale_fill_manual("Viral classification", values = colors)

psupp5a
psupp5b
```

# Supplementary Figure 6
```{r}
df_supp6 <- read.delim("/Users/chey2/Desktop/DOCK8_paper_codes/Input_files/Supp_6_input.txt", header = TRUE, sep = "\t", colClasses = c("factor", "factor", "factor", "numeric")) # input

df_supp6$Group <- factor(df_supp6$Group, levels = c("Pre-Tx", "Post-Tx-onIS-1M", "Post-Tx-onIS-2M", "Post-Tx-onIS-3M", "Post-Tx-onIS-6M", "Post-Tx-onIS>12M", "Post-Tx-offIS-12M", "Post-Tx-offIS-24M", "HVs", "Donor"))
df_supp6$kingdom <- factor(df_supp6$kingdom, levels = c("Bacteria", "Fungi", "Phage", "Eukaryotic virus"))

pie_each_site_plot_mean <- function(site){
  df_supp6 %>% 
    filter(site_specific=={{site}}) %>% 
    filter(!Group %in% c("Donor")) %>% 
    group_by(Group) %>% 
    mutate(ymax = cumsum(site_mean)) %>% 
    mutate(ymin = c(0, head(ymax, n=-1))) %>% 
    ungroup() -> pie_df
  
  p_out <- ggplot(pie_df, aes(ymax=ymax, ymin=ymin, xmax=1, xmin=0, fill=kingdom)) + geom_rect() + coord_polar(theta="y") + xlim(c(-0, 1)) + facet_grid(~Group) +
    scale_fill_manual("Classification", values = c("#31a253", "#3073b9","#c793c2","#5d488b")) +
    theme_bw() +
    theme(axis.text.x = element_blank(), axis.title.x=element_blank(), axis.text.y=element_blank(), axis.title.y=element_blank(), axis.ticks = element_blank()) +
    theme(panel.grid.minor=element_blank(), panel.grid.major=element_blank()) +
    theme(legend.position = "none") +
    theme(strip.text.x = element_blank())
  return(p_out)
}

Ra_venn_mean <- pie_each_site_plot_mean("Ra")
Mb_venn_mean <- pie_each_site_plot_mean("Mb")
Ac_venn_mean <- pie_each_site_plot_mean("Ac")
Vf_venn_mean <- pie_each_site_plot_mean("Vf")
Hp_venn_mean <- pie_each_site_plot_mean("Hp")
Ic_venn_mean <- pie_each_site_plot_mean("Ic")
Pc_venn_mean <- pie_each_site_plot_mean("Pc")
Ph_venn_mean <- pie_each_site_plot_mean("Ph")

venn_mean_list <- list(Ra_venn_mean, Mb_venn_mean, Ac_venn_mean, Vf_venn_mean, Hp_venn_mean, Ic_venn_mean, Pc_venn_mean, Ph_venn_mean)

psupp6 <- cowplot::plot_grid(plotlist = venn_mean_list, ncol = 1)
psupp6
```

# Supplementary Figure 7
```{r}
df_supp7 <- read.delim("/Users/chey2/Desktop/DOCK8_paper_codes/Input_files/Supp_7_input.txt", header = TRUE, sep = "\t", colClasses = c("factor", "factor", "factor", "factor", "numeric", "numeric")) # input

df_supp7$Genus <-  factor(df_supp7$Genus, levels = c("Alphatorquevirus", "Betatorquevirus", "Simplexvirus", "Roseolovirus", "Lymphocryptovirus","Cytomegalovirus", "Alphapapillomavirus", "Betapapillomavirus", "Gammapapillomavirus", "Mupapillomavirus", "Nupapillomavirus", "Alphapolyomavirus", "Betapolyomavirus", "Deltapolyomavirus", "Molluscipoxvirus"))

df_supp7$Species <- factor(df_supp7$Species, levels = c("Torque teno virus", "Alphatorquevirus homin10", "Torque teno virus 12", "Anelloviridae sp.", "Torque teno mini virus", "Betatorquevirus homini10", "Simplexvirus humanalpha1", "Roseolovirus humanbeta6b", "Lymphocryptovirus humangamma4", "Cytomegalovirus humanbeta5", "Alphapapillomavirus 4", "Alphapapillomavirus 1", "Alphapapillomavirus 2", "Alphapapillomavirus 3", "Alphapapillomavirus 10", "Alphapapillomavirus 14", "Alphapapillomavirus 8", "Alphapapillomavirus 9", "Betapapillomavirus 1", "Betapapillomavirus 2", "Betapapillomavirus 5", "Betapapillomavirus 3", "Betapapillomavirus 4", "Gammapapillomavirus 3", "Gammapapillomavirus 12", "Gammapapillomavirus 22", "Gammapapillomavirus 24", "Gammapapillomavirus 23", "Gammapapillomavirus 10", "Gammapapillomavirus 15", "Gammapapillomavirus 18", "Gammapapillomavirus 9", "Gammapapillomavirus sp", "Gammapapillomavirus 16", "Gammapapillomavirus 11", "Gammapapillomavirus 7", "Gammapapillomavirus 1", "Gammapapillomavirus 19", "Gammapapillomavirus 8", "Gammapapillomavirus 21", "Gammapapillomavirus 14", "Gammapapillomavirus 2", "Gammapapillomavirus 13", "Gammapapillomavirus", "Gammapapillomavirus 17", "Gammapapillomavirus 25", "Gammapapillomavirus 20", "Gammapapillomavirus 27", "Human Papilloma Virus", "Gammapapillomavirus 26", "Unclassified Gammapapillomavirus", "Gammapapillomavirus 5", "Mupapillomavirus 3", "Mupapillomavirus 1",
  "Mupapillomavirus 2", "Nupapillomavirus 1", "Alphapolyomavirus octihominis", "Alphapolyomavirus quintihominis", "Alphapolyomavirus nonihominis", "Betapolyomavirus hominis", "Betapolyomavirus secuhominis", "Betapolyomavirus tertihominis", "Betapolyomavirus quartihominis", "Deltapolyomavirus decihominis", "Deltapolyomavirus undecihominis", "Deltapolyomavirus sextihominis", "Deltapolyomavirus septihominis", "MOCV"))

df_supp7$grp <- factor(df_supp7$grp, levels = c("Pt01_Pre-Tx", "Pt02_Pre-Tx", "Pt03_Pre-Tx", "Pt04_Pre-Tx", "Pt05_Pre-Tx", "Pt06_Pre-Tx", "Pt07_Pre-Tx", "Pt08_Pre-Tx", "Pt09_Pre-Tx", "Pt10_Pre-Tx", "Pt11_Pre-Tx", "Pt12_Pre-Tx", "Pt13_Pre-Tx", "Pt14_Pre-Tx", "Pt15_Pre-Tx", "Pt16_Pre-Tx", "Pt17_Pre-Tx", "Pt18_Pre-Tx", "Pt19_Pre-Tx", "Pt20_Pre-Tx", "Pt21_Pre-Tx", "Pt22_Pre-Tx", "Pt23_Pre-Tx", "Pt24_Pre-Tx", "Pt20_Post-Tx-onIS-1M", "Pt22_Post-Tx-onIS-1M", "Pt23_Post-Tx-onIS-1M", "Pt20_Post-Tx-onIS-2M", "Pt22_Post-Tx-onIS-2M", "Pt23_Post-Tx-onIS-2M", "Pt18_Post-Tx-onIS-3M", "Pt19_Post-Tx-onIS-3M", "Pt20_Post-Tx-onIS-3M", "Pt21_Post-Tx-onIS-3M", "Pt22_Post-Tx-onIS-3M", "Pt23_Post-Tx-onIS-3M", "Pt24_Post-Tx-onIS-3M", "Pt15_Post-Tx-onIS-6M", "Pt16_Post-Tx-onIS-6M", "Pt17_Post-Tx-onIS-6M", "Pt18_Post-Tx-onIS-6M", "Pt19_Post-Tx-onIS-6M", "Pt20_Post-Tx-onIS-6M", "Pt21_Post-Tx-onIS-6M", "Pt09_Post-Tx-onIS>12M", "Pt12_Post-Tx-onIS>12M", "Pt13_Post-Tx-onIS>12M", "Pt05_Post-Tx-offIS-12M", "Pt06_Post-Tx-offIS-12M", "Pt07_Post-Tx-offIS-12M", "Pt08_Post-Tx-offIS-12M", "Pt10_Post-Tx-offIS-12M", "Pt11_Post-Tx-offIS-12M", "Pt14_Post-Tx-offIS-12M", "Pt15_Post-Tx-offIS-12M", "Pt16_Post-Tx-offIS-12M", "Pt17_Post-Tx-offIS-12M", "Pt18_Post-Tx-offIS-12M", "Pt19_Post-Tx-offIS-12M", "Pt20_Post-Tx-offIS-12M", "Pt01_Post-Tx-offIS-24M", "Pt02_Post-Tx-offIS-24M", "Pt03_Post-Tx-offIS-24M", "Pt04_Post-Tx-offIS-24M", "Pt05_Post-Tx-offIS-24M", "Pt06_Post-Tx-offIS-24M", "Pt07_Post-Tx-offIS-24M", "Pt08_Post-Tx-offIS-24M", "Pt09_Post-Tx-offIS-24M", "Pt10_Post-Tx-offIS-24M", "Pt11_Post-Tx-offIS-24M", "Pt24_Post-Tx-offIS-24M"))

df_supp7_mx <- df_supp7 %>% 
  select(Species, grp, relative) %>% 
  spread(grp, relative) %>% 
  replace(is.na(.), 0) %>% 
  column_to_rownames(var = "Species")

sapply(strsplit(colnames(df_supp7_mx), "_"), "[[", 1) -> temp_name
colnames(df_supp7_mx) <- temp_name
row_order <- levels(df_supp7$Species)
column_split = c(rep("Pre-Tx", 24), rep("Post-Tx-onIS-1M", 3), rep("Post-Tx-onIS-2M", 3), rep("Post-Tx-onIS-3M", 7), rep("Post-Tx-onIS-6M", 7), rep("Post-Tx-onIS>12M", 3), rep("Post-Tx-offIS-12M", 13), rep("Post-Tx-offIS-24M", 12))
row_split = c(rep("Anelloviridae", 6), rep("Herpesviridae", 4), rep("Papillomaviridae", 46), rep("Polyomaviridae", 11), rep("Poxviridae", 1))

col <- colorRamp2(c(0, 0.05, 0.1, 0.5, 10, 50, 100), c("#F7F7F7",  "#9970AB", "#C2A5CF", "#E7D4E8", "#A6DBA0", "#5AAE61", "#1B7837"))

row_ha <- rowAnnotation(block = anno_block(gp = gpar(fill = c("#DDE0C7", "#98D3BE", "#2B7FB7", "#D66027", "#FCD600"))), width = unit(1, "mm"))
row_ha_ha = rowAnnotation(pre = anno_barplot(rowSums(df_supp7_mx>0) %>% unname(), gp = gpar(fill=c(rep("#DDE0C7",6), rep("#98D3BE",4), rep("#2B7FB7",46), rep("#D66027",11), rep("#FCD600",1)))))


psupp7 <- Heatmap(as.matrix(df_supp7_mx),
                  row_order = row_order,
                  row_names_side = "left",
                  show_row_dend = F, 
                  show_column_dend = F,
                  col = col,
                  row_split = factor(row_split, levels = c("Anelloviridae", "Herpesviridae", "Papillomaviridae", "Polyomaviridae", "Poxviridae")),
                  row_gap = unit(1, "mm"),
                  column_gap = unit(1.5, "mm"),
                  row_title = NULL,
                  column_split = factor(column_split, levels = c("Pre-Tx", "Post-Tx-onIS-1M", "Post-Tx-onIS-2M", "Post-Tx-onIS-3M", "Post-Tx-onIS-6M", "Post-Tx-onIS>12M", "Post-Tx-offIS-12M", "Post-Tx-offIS-24M")),
                  cluster_row_slices = TRUE,
                  cluster_column_slices = FALSE, 
                  column_title = NULL, 
                  cluster_columns = FALSE, 
                  cluster_rows = FALSE,
                  left_annotation = row_ha,
                  right_annotation = row_ha_ha,
                  row_names_gp = gpar(fontsize = 6),
                  column_names_gp = gpar(fontsize = 6)) 

psupp7
```

# Supplementary Figure 9
```{r}
# need results from fig1h
# all sites and all groups in supplementary figure 9
p_pre_postoff12 <- map(target_taxa, paired_comparison, df1h_pre_postoff12, relative_abu)
p_pre_postoff24 <- map(target_taxa, paired_comparison, df1h_pre_postoff24, relative_abu)
p_pre_poston6 <- map(target_taxa, paired_comparison, df1h_pre_poston6, relative_abu)
p_pre_poston3 <- map(target_taxa, paired_comparison, df1h_pre_poston3, relative_abu)
p_pre_poston2 <- map(target_taxa, paired_comparison, df1h_pre_poston2, relative_abu)
p_pre_poston1 <- map(target_taxa, paired_comparison, df1h_pre_poston1, relative_abu)
p_pre_poston12 <- map(target_taxa, paired_comparison, df1h_pre_poston12, relative_abu)

p_on1 <- cowplot::plot_grid(plotlist = lapply(p_pre_poston1, "[[", 1), ncol = 1)
p_on2 <- cowplot::plot_grid(plotlist = lapply(p_pre_poston2, "[[", 1), ncol = 1)
p_on3 <- cowplot::plot_grid(plotlist = lapply(p_pre_poston3, "[[", 1), ncol = 1)
p_on6 <- cowplot::plot_grid(plotlist = lapply(p_pre_poston6, "[[", 1), ncol = 1)
p_on12 <- cowplot::plot_grid(plotlist = lapply(p_pre_poston12, "[[", 1), ncol = 1)
p_off12 <- cowplot::plot_grid(plotlist = lapply(p_pre_postoff12, "[[", 1), ncol = 1)
p_off24 <- cowplot::plot_grid(plotlist = lapply(p_pre_postoff24, "[[", 1), ncol = 1)
psupp9 <- plot_grid(p_on1, p_on2, p_on3, p_on6, p_on12, p_off12, p_off24, ncol=4, align="v")
```

# Supplementary Figure 10
```{r}
df_supp10 <- read.delim("/Users/chey2/Desktop/DOCK8_paper_codes/Input_files/Supp_10_input.txt", header = TRUE, sep = "\t", colClasses = c("factor", "factor", "numeric", "numeric", "numeric", "factor", "factor")) # input

pre_postoff_12 <- c("Pt11", "Pt05", "Pt10", "Pt07", "Pt08", "Pt06", "Pt16", "Pt15", "Pt14", "Pt18", "Pt17", "Pt19", "Pt20")
pre_postoff_24 <-  c("Pt01", "Pt11", "Pt02", "Pt09", "Pt05", "Pt03", "Pt04", "Pt10", "Pt07", "Pt08", "Pt06", "Pt24")
pre_poston6 <-  c("Pt16", "Pt15", "Pt18", "Pt17", "Pt21", "Pt19", "Pt20")
pre_poston3 <-  c("Pt18", "Pt21", "Pt24", "Pt22", "Pt19", "Pt20", "Pt23")
pre_poston2 <- c("Pt22", "Pt20", "Pt23")
pre_poston1 <- c("Pt22", "Pt20", "Pt23")
pre_poston_12 <- c("Pt09", "Pt12", "Pt13")

df_supp10 %>%
  filter(PID %in% pre_poston1) %>% 
  filter(Group %in% c("Pre-Tx", "Post-Tx-onIS-1M")) -> pre_poston1_num

df_supp10 %>%
  filter(PID %in% pre_poston2) %>% 
  filter(Group %in% c("Pre-Tx", "Post-Tx-onIS-2M")) -> pre_poston2_num

df_supp10 %>%
  filter(PID %in% pre_poston3) %>% 
  filter(Group %in% c("Pre-Tx", "Post-Tx-onIS-3M")) -> pre_poston3_num

df_supp10 %>%
  filter(PID %in% pre_poston6) %>% 
  filter(Group %in% c("Pre-Tx", "Post-Tx-onIS-6M")) -> pre_poston6_num

df_supp10 %>%
  filter(PID %in% pre_postoff_12) %>% 
  filter(Group %in% c("Pre-Tx", "Post-Tx-offIS-12M")) -> pre_postoff12_num

df_supp10 %>%
  filter(PID %in% pre_postoff_24) %>% 
  filter(Group %in% c("Pre-Tx", "Post-Tx-offIS-24M")) -> pre_postoff24_num


dabest_prepare <- function(df, grp=c("Pre-Tx-1M", "Pre-Tx-2M", "Pre-Tx-3M", "Pre-Tx-6M", "Pre-Tx-12M", "Pre-Tx-24M")){
  df1 <- {{df}} %>% 
    mutate(grp=case_when(
       Group=="Pre-Tx" ~ {{grp}},
       TRUE ~ Group
    )) %>% 
    select(-Group)
  return(df1)
}

pre_poston1_num_df <- dabest_prepare(pre_poston1_num, grp="Pre-Tx-1M")
pre_poston2_num_df <- dabest_prepare(pre_poston2_num, grp="Pre-Tx-2M")
pre_poston3_num_df <- dabest_prepare(pre_poston3_num, grp="Pre-Tx-3M")
pre_poston6_num_df <- dabest_prepare(pre_poston6_num, grp="Pre-Tx-6M")
pre_postoff12_num_df <- dabest_prepare(pre_postoff12_num, grp="Pre-Tx-12M")
pre_postoff24_num_df <- dabest_prepare(pre_postoff24_num, grp="Pre-Tx-24M")

hpv_type_num_list <- rbind(pre_poston1_num_df, pre_poston2_num_df, pre_poston3_num_df, pre_poston6_num_df, pre_postoff12_num_df, pre_postoff24_num_df)

hpv_type_multi_groups <- load(hpv_type_num_list,
                     x = grp, y = normalized_num,
                     idx = list(
                         c("Pre-Tx-1M", "Post-Tx-onIS-1M"),
                         c("Pre-Tx-2M", "Post-Tx-onIS-2M"),
                         c("Pre-Tx-3M", "Post-Tx-onIS-3M"),
                         c("Pre-Tx-6M", "Post-Tx-onIS-6M"),
                         c("Pre-Tx-12M", "Post-Tx-offIS-12M"),
                         c("Pre-Tx-24M", "Post-Tx-offIS-24M")))

psupp10 <- hpv_type_multi_groups %>%
  mean_diff() %>%
  dabest_plot(show_zero_dot=FALSE, raw_marker_size=0.5, tufte_size=0.3, es_marker_size=0.5, es_line_size=0.3, raw_marker_spread=1.5, swarm_x_text=6, swarm_y_text=6, contrast_x_text=6, contrast_y_text=6, raw_marker_alpha=0.5, swarm_label="Number of normalized HPV types (log)", contrast_label="Mean difference", custom_palette="igv")

psupp10
```

# Supplementary Figure 11
```{r}
# need results from fig1i
psupp11 <- df1i %>% 
  ggplot(aes(x=group, y=mean_fold, group=kingdom, color =kingdom)) +
  geom_point(size=0.7) +
  geom_line(size=0.5) +
  facet_grid(site_specific~.) +
  geom_errorbar(aes(ymin=mean_fold-sd, ymax=mean_fold+sd), width=.3, linewidth=.2) +
  scale_color_manual("Classification", values = c("#31a253", "#3073b9","#5d488b")) +
  theme_bw() +
  theme(axis.text.x = element_text(angle = 90, size=8, hjust=1, vjust=0.3), axis.title.x=element_blank(), axis.text.y=element_text(size=8), axis.title.y=element_text(size=10)) +
  theme(panel.grid.minor=element_blank(), panel.grid.major=element_blank(), legend.title=element_text(size=10), legend.text=element_text(size=8)) + 
  ylab("Genome equivalents log10 fold change")
```

# Supplementary Figure 12
```{r}
df_supp12 <- read.delim("/Users/chey2/Desktop/DOCK8_paper_codes/Input_files/Supp_12_input.txt", header = TRUE, sep = "\t", colClasses = c("factor", "numeric", "numeric")) # input
df_supp12$group <- factor(df_supp12$group, levels = c(df_supp12$group))

model <- lm(p50 ~ group_num, data = df_supp12)
segmented_model <- segmented(model, seg.Z = ~group_num, psi = 5)
df_supp12$predicted <- predict(segmented_model)
breakpoint <- segmented_model$psi[2]

psupp12 <- ggplot(df_supp12, aes(x = group, y = p50)) +
  geom_point(size = 1, color = "black") +  # Original data points
  geom_line(aes(y = predicted, group = 1), color = "blue", size = 0.5) +  # Fitted piecewise regression line
  geom_vline(xintercept = as.numeric(breakpoint), linetype = "dashed", color = "red", size = 0.5) +  # Breakpoint line
  labs(title = "Piecewise regression of top virus abundance",
       y = "Median relative abundance (%)") +
  theme(
    panel.background = element_blank(),  # Remove background
    axis.line = element_line(color = "black"),  # Add axis lines
    plot.title = element_text(hjust = 0.5, face = "bold", size = 12),  # Center and bold title
    axis.title.x = element_blank(),
    axis.title.y = element_text(face = "bold", size = 10),  # Bold axis titles
    axis.text = element_text(color = "black")  # Black axis text
  )

psupp12
```

# Supplementary Figure 13
```{r}
df_supp13 <- read.delim("/Users/chey2/Desktop/DOCK8_paper_codes/Input_files/Supp_13_1_input.txt", header = TRUE, sep = "\t") # input
df_supp13$Group <- factor(df_supp13$Group, levels = c("Pre-Tx", "Post-Tx-onIS-1M", "Post-Tx-onIS-2M", "Post-Tx-onIS-3M"))


plot_for_each_viruses <- function(tax){
  df_supp13 %>% 
  filter(Ref=={{tax}}) %>% 
  #select(-Ref) %>% 
  spread(Group, Cov) %>% 
  replace(is.na(.), 0) %>% 
  column_to_rownames(var = "site_specific") -> df_supp13_mx
  df_supp13_mx_input <- log(df_supp13_mx+1)
  col <- colorRamp2(c(0, 5, 10, 15), c("#DEEBF7", "#7FCDBB", "#41B6C4", "#1D91C0"))
  p_pecific <- Heatmap(as.matrix(df_supp13_mx_input),
        row_order = c("Ra", "Mb", "Vf", "Hp", "Ac", "Ic", "Pc", "Ph"),
        row_names_side = "left",
        show_row_dend = F, 
        show_column_dend = F,
        col = col,
        cluster_row_slices = FALSE,
        cluster_column_slices = FALSE, 
        column_title = NULL, 
        cluster_columns = FALSE, 
        cluster_rows = FALSE,
        row_names_gp = gpar(fontsize = 6),
        column_names_gp = gpar(fontsize = 6)) 
  return(p_pecific)
}

p_WuPyv <- plot_for_each_viruses("DOCK8Poly_11")
p_TTV <- plot_for_each_viruses("DOCK8Ane_6")
p_HPV116 <- plot_for_each_viruses("DOCK8HPV_71")

p_WuPyv
p_TTV
p_HPV116
```

# Supplementary Figure 14a
```{r}
df_supp14 <- read.delim("/Users/chey2/Desktop/DOCK8_paper_codes/Input_files/Supp_14_input.txt", header = TRUE, sep = "\t") # input

df_supp14 %>% 
  group_by(subject_id, taxa, Group) %>% 
  summarise(abu = sum(relative_abu)) %>% 
  ungroup() -> df_supp14_f

df_supp14_f$taxa <- factor(df_supp14_f$taxa, levels = c("Bacteria", "Actinobacteria", "Micrococcus luteus", "Corynebacterium tuberculostearicum", "Cutibacterium acnes", "Firmicutes", "Staphylococcus hominis", "Staphylococcus epidermidis","Staphylococcus aureus", "Proteobacteria", "Alphaproteobacteria",  "Gammaproteobacteria", "Fungi", "Malassezia restricta", "Malassezia globosa", "Eukaryotic virus", "Gammapapillomavirus 9", "Gammapapillomavirus 24", "Gammapapillomavirus 8", "Gammapapillomavirus 22", "Betapapillomavirus 2", "Betapapillomavirus 1", "Molluscum contagiosum virus"))
df_supp14_f$subject_id <- factor(df_supp14_f$subject_id, levels = c("Pt01", "Pt02", "Pt03", "Pt04", "Pt05", "Pt06", "Pt07", "Pt08", "Pt09",  "Pt10", "Pt11", "Pt12", "Pt13",  "Pt14", "Pt15", "Pt16", "Pt17", "Pt18", "Pt19", "Pt20", "Pt21", "Pt22", "Pt23", "Pt24", "Hv01", "Hv02", "Hv03", "Hv04", "Hv05", "Hv06"))


color_pall_comb <- c(brewer.pal(5, 'Greens'),
               brewer.pal(5, 'Oranges')[c(1,3, 4, 5)],
               brewer.pal(11, 'BrBG')[5:3],
               brewer.pal(11, "PiYG")[5:3],
               brewer.pal(5, 'Purples'),
               brewer.pal(9, 'Blues')[7:8],
               brewer.pal(9, 'Blues')[3]
               )

psupp14 <- df_supp14_f %>% 
  filter(!Group=="Donor") %>% 
  ggplot(aes(x=subject_id, y = abu, fill = taxa)) +
  geom_bar(stat = "identity", position = "fill", width = 0.8) +
  scale_y_continuous(labels = function(x) x*100) +
  theme_bw() +
  facet_grid(~factor(Group, levels = c("Pre-Tx", "Post-Tx-onIS-1M", "Post-Tx-onIS-2M", "Post-Tx-onIS-3M", "Post-Tx-onIS-6M", "Post-Tx-onIS>12M", "Post-Tx-offIS-12M", "Post-Tx-offIS-24M", "HVs")), scales = "free_x",space = "free_x") + 
   scale_fill_manual("Classification", values = color_pall_comb) +
    theme(axis.text.x = element_text(angle = 90, size=7, hjust=1, vjust=0.3),axis.title.x=element_blank(), axis.text.y=element_text(size=7), axis.title.y=element_text(size=10)) +
  theme(panel.grid.minor=element_blank(), panel.grid.major=element_blank(), legend.box = "vertical", legend.title=element_text(size=10), legend.text=element_text(size=7), legend.key.size=unit(0.5, "cm")) + 
  theme(strip.text.x=element_text(size=8), strip.text.y=element_text(size=8)) +
  ylab("Mean relative abundance (%)") +
  guides(fill=guide_legend(ncol=1))

psupp14
```

# Supplementary Figure 14b
```{r}
df_supp14b <- read.delim("/Users/chey2/Desktop/DOCK8_paper_codes/Input_files/Supp_14b_input.txt", header = TRUE, sep = "\t") # input

df_supp14b$final_taxa <- factor(df_supp14b$final_taxa, levels = c("Bacteria", "Actinobacteriota", "Micrococcus luteus", "Corynebacterium tuberculostearicum", "Cutibacterium acnes", "Firmicutes", "Staphylococcus hominis", "Staphylococcus epidermidis", "Staphylococcus aureus", "Alphaproteobacteria", "Gammaproteobacteria"))

df_supp14b$subject_id <- factor(df_supp14b$subject_id, levels = c("Pt01", "Pt02", "Pt03", "Pt04", "Pt05", "Pt06", "Pt07", "Pt08", "Pt09", "Pt10", "Pt11", "Pt12", "Pt13", "Pt14", "Pt15", "Pt16", "Pt17", "Pt18", "Pt19", "Pt20", "Pt21", "Pt22", "Pt23", "Pt24", "Hv01", "Hv02", "Hv03", "Hv04", "Hv05", "Hv06"))

df_supp14b %>% 
  group_by(subject_id,final_taxa,Group) %>% 
  summarise(new_abu=sum(rela_abu)) %>% 
  ungroup() -> df_supp14b_p

my_color_16s <- c(brewer.pal(5, 'Greens'),
                   brewer.pal(5, 'Oranges')[c(1,3, 4, 5)],
                   brewer.pal(11, 'BrBG')[4:3])

df_supp14b$Group <- factor(df_supp14b$Group, levels = c("Pre-Tx", "Post-Tx-onIS-1M", "Post-Tx-onIS-2M", "Post-Tx-onIS-3M", "Post-Tx-onIS-6M", "Post-Tx-onIS>12M", "Post-Tx-offIS-12M", "Post-Tx-offIS-24M", "HVs"))

psupp14b <- df_supp14b_p %>%
    ggplot(aes(x=subject_id, y = new_abu, fill = factor(final_taxa))) +
    geom_bar(stat = "identity", position = "fill", width = 0.8) +
    scale_y_continuous(labels = function(x) x*100) +
    facet_grid(~Group, scales = "free_x", space = "free_x") +
    theme_bw() +
    theme(axis.text.x = element_text(angle = 90, size=7, hjust=1, vjust=0.3), axis.title.x=element_blank(), axis.text.y=element_text(size=7), axis.title.y=element_text(size=10)) +
  theme(panel.grid.minor=element_blank(), panel.grid.major=element_blank(), legend.title=element_text(size=10), legend.text=element_text(size=7)) +
  ylab("Mean relative abundance (%)") +
    scale_fill_manual("Classification", values = my_color_16s)

psupp14b
```

# Supplementary Figure 14c
```{r}
df_supp14c <- read.delim("/Users/chey2/Desktop/DOCK8_paper_codes/Input_files/Supp_14c_input.txt", header = TRUE, sep = "\t") # input

df_supp14c$site_specific <- factor(df_supp14c$site_specific, levels = c("Ra", "Mb", "Vf", "Hp", "Ac", "Ic", "Pc", "Ph"))
df_supp14c$Group <- factor(df_supp14c$Group, levels = c("Pre-Tx", "Post-Tx-onIS-1M", "Post-Tx-onIS-2M", "Post-Tx-onIS-3M", "Post-Tx-onIS-6M", "Post-Tx-offIS-12M", "Post-Tx-offIS-24M", "HVs"))

psupp14c <- df_supp14c %>%
  ggplot(aes(x=Group, y=Shannon)) +
  facet_grid(site_specific~.) +
  geom_boxplot(lwd = 0.15, outlier.size = 0.1, outlier.shape = NA) +
  theme_bw() +
  theme(axis.title.x=element_blank(), axis.text.x = element_text(size = 8, angle = 45, hjust = 0.98, vjust = 0.98), axis.title.y=element_text(size=10)) +
  theme(axis.text.y = element_text(size = 8)) +
  ylab("Shannon") +
  theme(panel.grid.minor=element_blank(), panel.grid.major=element_blank(), legend.title=element_text(size=8)) + theme(legend.text = element_text(size = 8))

psupp14c
```

# Supplementary Figure 14d
```{r}
df_supp14d <- read.delim("/Users/chey2/Desktop/DOCK8_paper_codes/Input_files/Supp_14d_input.txt", header = TRUE, sep = "\t") # input

df_supp14d$new_grp <- factor(df_supp14d$new_grp, levels = c("Pre-Tx", "Post-Tx-onIS-1M", "Post-Tx-onIS-2M", "Post-Tx-onIS-3M", "Post-Tx-onIS-6M", "Post-Tx-offIS-12M", "Post-Tx-offIS-24M"))

df_supp14d %>% 
  filter(S %in% c("Cutibacterium acnes", "Staphylococcus epidermidis", "Staphylococcus aureus")) -> df_supp14d_three

df_supp14d_three$S <- factor(df_supp14d_three$S, levels = c("Cutibacterium acnes", "Staphylococcus epidermidis", "Staphylococcus aureus"))

df_supp14d_three$site_specific <- factor(df_supp14d_three$site_specific, levels = c("Ra", "Mb", "Vf", "Hp", "Ac", "Ic", "Pc", "Ph"))

psupp14d <- df_supp14d_three %>% 
    ggplot(aes(x=new_grp, y=mean, group=S, color =S)) +
    geom_point(size=0.7) +
    geom_line(size=0.3) +
    facet_grid(site_specific~.) +
    geom_errorbar(aes(ymin=mean-sd, ymax=mean+sd), width=.3, linewidth=.1) +
    scale_color_manual("Species", values = c("#006d2c", "#E6550D", "#A63603")) +
    theme_bw() +
    theme(axis.text.x = element_text(angle = 45, size=8, hjust=0.95,vjust=1), axis.title.x=element_blank(), axis.text.y=element_text(size=8), axis.title.y=element_text(size=10)) +
    theme(panel.grid.minor=element_blank(), panel.grid.major=element_blank(), legend.title=element_text(size=10), legend.text=element_text(size=8)) + 
    ylab("Mean relative abundance log10 fold change")

psupp14d
```

# Supplementary Figure 14e
```{r}
df_supp14e <- read.delim("/Users/chey2/Desktop/DOCK8_paper_codes/Input_files/Supp_14e_input.txt", header = TRUE, sep = "\t") # input

dabest_plot_fun_ASV <- function(site){
  df_supp14e %>% 
    filter(site_specific=={{site}}) -> ASV_all_d_temp
  multi_groups <- load(ASV_all_d_temp,
                     x = grp, y = ASVcount,
                     idx = list(
                       c("Pre-Tx-1M", "Post-Tx-onIS-1M"),
                       c("Pre-Tx-2M", "Post-Tx-onIS-2M"),
                       c("Pre-Tx-3M", "Post-Tx-onIS-3M"),
                       c("Pre-Tx-6M", "Post-Tx-onIS-6M"),
                       c("Pre-Tx-12M", "Post-Tx-offIS-12M"),
                       c("Pre-Tx-24M", "Post-Tx-offIS-24M")))
  p <- multi_groups %>%
  mean_diff() %>%
  dabest_plot(show_zero_dot=FALSE, raw_marker_size=0.5, tufte_size=0.3, es_marker_size=0.5, es_line_size=0.3, raw_marker_spread=1.5, swarm_x_text=6, swarm_y_text=6, contrast_x_text=6, contrast_y_text=6, raw_marker_alpha=0.5, swarm_label="Number of unique ASV", contrast_label="Mean difference", custom_palette="igv")
  return(p)
}

poston1_ASV <- prepare_paired_samples(c("Pre-Tx", "Post-Tx-onIS-1M"))
poston2_ASV <- prepare_paired_samples(c("Pre-Tx", "Post-Tx-onIS-2M"))
poston3_ASV <- prepare_paired_samples(c("Pre-Tx", "Post-Tx-onIS-3M"))
poston6_ASV <- prepare_paired_samples(c("Pre-Tx", "Post-Tx-onIS-6M"))
postoff12_ASV <- prepare_paired_samples(c("Pre-Tx", "Post-Tx-offIS-12M"))
postoff24_ASV <- prepare_paired_samples(c("Pre-Tx", "Post-Tx-offIS-24M"))

dabest_prepare <- function(df, grp=c("Pre-Tx-1M", "Pre-Tx-2M", "Pre-Tx-3M", "Pre-Tx-6M", "Pre-Tx-12M", "Pre-Tx-24M")){
  df1 <- {{df}} %>% 
    mutate(grp=case_when(
       Group=="Pre-Tx" ~ {{grp}},
       TRUE ~ Group
    )) %>% 
    select(-Group)
  return(df1)
}

poston1_ASV_df <- dabest_prepare(poston1_ASV, grp="Pre-Tx-1M") 
poston2_ASV_df <- dabest_prepare(poston2_ASV, grp="Pre-Tx-2M") 
poston3_ASV_df <- dabest_prepare(poston3_ASV, grp="Pre-Tx-3M") 
poston6_ASV_df <- dabest_prepare(poston6_ASV, grp="Pre-Tx-6M") 
postoff12_ASV_df <- dabest_prepare(postoff12_ASV, grp="Pre-Tx-12M") 
postoff24_ASV_df <- dabest_prepare(postoff24_ASV, grp="Pre-Tx-24M") 

ASV_all_d <- rbind(poston1_ASV_df, poston2_ASV_df, poston3_ASV_df, poston6_ASV_df, postoff12_ASV_df, postoff24_ASV_df)

dabest_plot_fun_ASV <- function(site){
  ASV_all_d %>% 
    filter(site_specific=={{site}}) -> ASV_all_d_temp
  #ASV_all_d_temp$site_specific <- droplevels(ASV_all_d_temp$site_specific)
  multi_groups <- load(ASV_all_d_temp,
                     x = grp, y = ASVcount,
                     idx = list(
                       c("Pre-Tx-1M", "Post-Tx-onIS-1M"),
                       c("Pre-Tx-2M", "Post-Tx-onIS-2M"),
                       c("Pre-Tx-3M", "Post-Tx-onIS-3M"),
                       c("Pre-Tx-6M", "Post-Tx-onIS-6M"),
                       c("Pre-Tx-12M", "Post-Tx-offIS-12M"),
                       c("Pre-Tx-24M", "Post-Tx-offIS-24M")))
  p <- multi_groups %>%
  mean_diff() %>%
  dabest_plot(show_zero_dot=FALSE, raw_marker_size=0.5, tufte_size=0.3, es_marker_size=0.5, es_line_size=0.3, raw_marker_spread=1.5, swarm_x_text=6, swarm_y_text=6, contrast_x_text=6, contrast_y_text=6, raw_marker_alpha=0.5, swarm_label="Number of unique ASV", contrast_label="Mean difference", custom_palette="igv")
  return(p)
}


site_lt <- c("Ra", "Mb", "Vf", "Ac", "Ic", "Pc", "Ph")
all_figures_7_ASV <- map(site_lt, dabest_plot_fun_ASV)

ASV_all_d %>% 
  filter(site_specific=="Hp") %>% 
  filter(!grp %in% c("Pre-Tx-1M", "Post-Tx-onIS-1M", "Pre-Tx-2M", "Post-Tx-onIS-2M")) -> ASV_all_Hp

ASV_all_Hp_dabest <- load(ASV_all_Hp,
                     x = grp, y = ASVcount,
                     idx = list(
                       c("Pre-Tx-3M", "Post-Tx-onIS-3M"),
                       c("Pre-Tx-6M", "Post-Tx-onIS-6M"),
                       c("Pre-Tx-12M", "Post-Tx-offIS-12M"),
                       c("Pre-Tx-24M", "Post-Tx-offIS-24M")))

psupp14e_1 <- ASV_all_Hp_dabest %>%
  mean_diff() %>%
  dabest_plot(show_zero_dot=FALSE, raw_marker_size=0.5, tufte_size=0.3, es_marker_size=0.5, es_line_size=0.3, raw_marker_spread=1.5, swarm_x_text=6, swarm_y_text=6, contrast_x_text=6, contrast_y_text=6, raw_marker_alpha=0.5, swarm_label="Number of unique ASV", contrast_label="Mean difference", custom_palette="igv")

psupp14e_7 <- cowplot::plot_grid(plotlist = all_figures_7_ASV)
psupp14e_1
```

# Supplementary Figure 14f
```{r}
df_supp14f <- read.delim("/Users/chey2/Desktop/DOCK8_paper_codes/Input_files/Supp_14f_input.txt", header = TRUE, sep = "\t") # input

df_supp14f$Group_Comparison <- factor(df_supp14f$Group_Comparison, levels = c("Post-Tx-onIS-1M vs Pre-Tx","Post-Tx-onIS-2M vs Pre-Tx", "Post-Tx-onIS-3M vs Pre-Tx", "Post-Tx-onIS-6M vs Pre-Tx", "Post-Tx-offIS-12M vs Pre-Tx", "Post-Tx-offIS-24M vs Pre-Tx"))
df_supp14f$Group1 <- factor(df_supp14f$Group1, levels = c("mean_pre", "mean_post"))
df_supp14f$Site_Specific <- factor(df_supp14f$Site_Specific, levels = c("Ra", "Mb", "Vf", "Hp", "Ac", "Ic", "Pc", "Ph"))
df_supp14f[is.na(df_merged)] = 0

psupp14f <- df_supp14f %>% 
  ggplot(aes(x=Group1, y=Mean, fill=Group1)) + 
  geom_bar(stat="identity") + 
  facet_grid(Site_Specific~Group_Comparison) +
  geom_errorbar(aes(x=Group1, ymin=Mean-Sd, ymax=Mean+Sd), width=0.1, colour="black", alpha=0.3) +
  theme_bw() + 
  theme(axis.text.x = element_blank(), axis.title.x=element_blank(), axis.text.y=element_text(size=8), axis.title.y=element_text(size=10)) + 
  theme(panel.grid.minor=element_blank(), panel.grid.major=element_blank(), legend.title=element_text(size=10), legend.text=element_text(size=8)) +
  scale_fill_manual(name = "Sample", 
                    labels = c("Paired pre-hsct", "Paired post-hsct"),
                    values = c("#d8fdc1", "#54baba")) +
  coord_cartesian(ylim = c(0, 1)) +
  ylab("Proportion of shared species (Read-Weighted)") 

psupp14f
```

# Supplementary Figure 15
```{r}
df_supp15 <- read.delim2(file = "/Users/chey2/Desktop/DOCK8_paper_codes/Input_files/Supp_15_input.txt", header = TRUE, sep = "\t", stringsAsFactors = FALSE, check.names = FALSE)

df_supp15$relative_abu <- as.numeric(df_supp15$relative_abu)

all_samples <- c("Pt05 Post-Tx-offIS-12M", "Pt06 Post-Tx-offIS-12M", "Pt07 Post-Tx-offIS-12M", "Pt08 Post-Tx-offIS-12M", "Pt10 Post-Tx-offIS-12M", "Pt11 Post-Tx-offIS-12M", "Pt14 Post-Tx-offIS-12M", "Pt15 Post-Tx-offIS-12M", "Pt16 Post-Tx-offIS-12M", "Pt17 Post-Tx-offIS-12M", "Pt18 Post-Tx-offIS-12M",  "Pt19 Post-Tx-offIS-12M", "Pt20 Post-Tx-offIS-12M", "Pt01 Post-Tx-offIS-24M", "Pt02 Post-Tx-offIS-24M", "Pt03 Post-Tx-offIS-24M", "Pt04 Post-Tx-offIS-24M", "Pt05 Post-Tx-offIS-24M", "Pt06 Post-Tx-offIS-24M", "Pt07 Post-Tx-offIS-24M", "Pt08 Post-Tx-offIS-24M", "Pt09 Post-Tx-offIS-24M", "Pt10 Post-Tx-offIS-24M", "Pt11 Post-Tx-offIS-24M", "Pt24 Post-Tx-offIS-24M", "Hv01 HVs", "Hv02 HVs", "Hv03 HVs", "Hv04 HVs",  "Hv05 HVs", "Hv06 HVs")

all_Malassezia_species <- df_supp15$Ref %>% unique()

# define function to plot Malassezia heatmap
heatmap_for_Malassezia <- function(site){
  df_supp15 %>% 
    filter(site_specific=={{site}}) %>% 
    select(-site_specific) -> df_Malassezia_site
  species_lst <- all_Malassezia_species[!all_Malassezia_species %in% df_Malassezia_site$Ref]
  for (i in species_lst) {
  df_Malassezia_site <- df_Malassezia_site %>% add_row(grp="Pt01 Post-Tx-offIS-24M",Ref={{i}}, relative_abu=0) 
  }
  # add subjects
  sample_lst <- all_samples[!all_samples %in% df_Malassezia_site$grp]
  species_temp <- df_Malassezia_site$Ref[1]
  for (i in sample_lst) {
  df_Malassezia_site <- df_Malassezia_site %>% 
    add_row(grp={{i}},Ref=species_temp, relative_abu=0)
  }
  df_Malassezia_site$grp <- factor(df_Malassezia_site$grp, levels = all_samples)
  df_Malassezia_site %>% 
    arrange(factor(df_Malassezia_site$grp)) -> df_Malassezia_site
  
  df_Malassezia_site$grp <- factor(df_Malassezia_site$grp, levels = )
  
  df_Malassezia_site %>% 
    pivot_wider(names_from = grp, values_from = relative_abu) %>%
    replace(is.na(.), 0) %>% 
    mutate(Ref=factor(Ref, levels = c("Malassezia_palmae", "Malassezia_rara", "Malassezia_auris", "Malassezia_furfur",
                                      "Malassezia_globosa", "Malassezia_restricta", "Malassezia_arunalokei", "Malassezia_sympodialis",
                                      "Malassezia_dermatis", "Malassezia_obtusa", "Malassezia_slooffiae"))) %>%
    arrange(Ref) %>% 
    column_to_rownames(var = "Ref") -> df_Malassezia_site_heatmap
  
  column_split_sample = c(rep("Post-Tx-offIS-12M", 13), rep("Post-Tx-offIS-24M", 12), rep("HVs", 6))
  sapply(strsplit(colnames(df_Malassezia_site_heatmap), " "), "[[", 1) -> temp_name_sample
  colnames(df_Malassezia_site_heatmap) <- temp_name_sample
  
  col_fun = colorRamp2(c(0, 3, 5, 10, 20, 30, 90), c("#F5F5F5", "#F9D8A3", "#FDF1D1", "#E6F598", "#ABDDA4", "#66C2A5", "#3288BD"))
  
  p_Malassezia <- Heatmap(as.matrix(df_Malassezia_site_heatmap),
                               row_order = rownames(df_Malassezia_site_heatmap),
                               row_names_side = "left",
                               show_row_dend = F, 
                               show_column_dend = F,
                               col = col_fun,
                               column_gap = unit(1.5, "mm"),
                               row_title = NULL,
                               column_split = factor(column_split_sample, levels = c("Post-Tx-offIS-12M", "Post-Tx-offIS-24M", "HVs")),
                               cluster_row_slices = TRUE,
                               cluster_column_slices = FALSE, 
                               column_title = NULL, 
                               cluster_columns = FALSE, 
                               cluster_rows = FALSE,
                               row_names_gp = gpar(fontsize = 6),
                               column_names_gp = gpar(fontsize = 6),
                               heatmap_legend_param = list(title = "Relative abundance (%)", at = c(0, 20, 40, 60, 80, 100)))
  return(p_Malassezia)
}

Ra_Malassezia <- heatmap_for_Malassezia("Ra")
Mb_Malassezia <- heatmap_for_Malassezia("Mb")
Vf_Malassezia <- heatmap_for_Malassezia("Vf")
Hp_Malassezia <- heatmap_for_Malassezia("Hp")
Ac_Malassezia <- heatmap_for_Malassezia("Ac")
Ic_Malassezia <- heatmap_for_Malassezia("Ic")
Pc_Malassezia <- heatmap_for_Malassezia("Pc")
Ph_Malassezia <- heatmap_for_Malassezia("Ph")

Ra_Malassezia
Mb_Malassezia
Vf_Malassezia
Hp_Malassezia
Ac_Malassezia
Ic_Malassezia
Pc_Malassezia
Ph_Malassezia
```

# Supplementary Figure 16
```{r}
# need function from figure3a
psupp16 <- map(l_sites, topn=4, Pcoa_for_each_site, variable_scalar = 2.7, label_s = 0.3, arrow = TRUE, HV = TRUE, pairwise_PERMANOVA_ctrl_func = TRUE)
psupp16
```

# Supplementary Figure 17b
```{r}
df_supp17 <- read.delim2(file = "/Users/chey2/Desktop/DOCK8_paper_codes/Input_files/Supp_17_input.txt", header = TRUE, sep = "\t", stringsAsFactors = FALSE, check.names = FALSE)

p_coryne <- function(species){
  df_coryne %>%
    filter(Taxa=={{species}}) %>% 
    select(-Taxa) -> df_coryne_target
  all_comb <- df_coryne$group %>% unique()
  need_add_zero_row_coryne <- all_comb[!all_comb %in% df_coryne_target$group]
  
  for (i in need_add_zero_row_coryne) {
    df_coryne_target <- df_coryne_target %>% 
      add_row(site_specific="Ac", group={{i}}, relab=0)}
  
  df_coryne_target$group <- factor(df_coryne_target$group, levels = c("Pt01 Pre-Tx", "Pt02 Pre-Tx", "Pt03 Pre-Tx", "Pt04 Pre-Tx", "Pt05 Pre-Tx", "Pt06 Pre-Tx", "Pt07 Pre-Tx", "Pt08 Pre-Tx", "Pt09 Pre-Tx", "Pt10 Pre-Tx", "Pt11 Pre-Tx", "Pt12 Pre-Tx", "Pt13 Pre-Tx", "Pt14 Pre-Tx", "Pt15 Pre-Tx", "Pt16 Pre-Tx", "Pt17 Pre-Tx", "Pt18 Pre-Tx", "Pt19 Pre-Tx", "Pt20 Pre-Tx", "Pt21 Pre-Tx", "Pt22 Pre-Tx", "Pt23 Pre-Tx", "Pt24 Pre-Tx","Pt18 Post-Tx-onIS≤3M", "Pt19 Post-Tx-onIS≤3M", "Pt20 Post-Tx-onIS≤3M", "Pt21 Post-Tx-onIS≤3M", "Pt22 Post-Tx-onIS≤3M","Pt23 Post-Tx-onIS≤3M", "Pt24 Post-Tx-onIS≤3M", "Pt15 Post-Tx-onIS-6M", "Pt16 Post-Tx-onIS-6M", "Pt17 Post-Tx-onIS-6M", "Pt18 Post-Tx-onIS-6M", "Pt19 Post-Tx-onIS-6M", "Pt20 Post-Tx-onIS-6M", "Pt21 Post-Tx-onIS-6M", "Pt09 Post-Tx-onIS>12M", "Pt12 Post-Tx-onIS>12M", "Pt13 Post-Tx-onIS>12M","Pt05 Post-Tx-offIS-12M", "Pt06 Post-Tx-offIS-12M", "Pt07 Post-Tx-offIS-12M", "Pt08 Post-Tx-offIS-12M", "Pt10 Post-Tx-offIS-12M", "Pt11 Post-Tx-offIS-12M", "Pt14 Post-Tx-offIS-12M", "Pt15 Post-Tx-offIS-12M", "Pt16 Post-Tx-offIS-12M", "Pt17 Post-Tx-offIS-12M", "Pt18 Post-Tx-offIS-12M", "Pt19 Post-Tx-offIS-12M", "Pt20 Post-Tx-offIS-12M", "Pt01 Post-Tx-offIS-24M", "Pt02 Post-Tx-offIS-24M", "Pt03 Post-Tx-offIS-24M", "Pt04 Post-Tx-offIS-24M", "Pt05 Post-Tx-offIS-24M", "Pt06 Post-Tx-offIS-24M", "Pt07 Post-Tx-offIS-24M", "Pt08 Post-Tx-offIS-24M", "Pt09 Post-Tx-offIS-24M", "Pt10 Post-Tx-offIS-24M", "Pt11 Post-Tx-offIS-24M", "Pt24 Post-Tx-offIS-24M", "Hv01 HVs", "Hv02 HVs", "Hv03 HVs", "Hv04 HVs", "Hv05 HVs", "Hv06 HVs"))
  df_coryne_target %>% 
    arrange(factor(group)) -> df_coryne_target_plot
  df_coryne_target_plot$site_specific <- factor(df_coryne_target_plot$site_specific, levels = c("Ra", "Mb", "Vf", "Hp", "Ac", "Ic", "Pc", "Ph"))
  df_coryne_target_plot$relab <- df_coryne_target_plot$relab*100
  
  df_coryne_target_plot %>%
    pivot_wider(names_from = group, values_from = relab) %>%
    replace(is.na(.), 0) %>% 
    mutate(site_specific=factor(site_specific, levels = c("Ra", "Mb", "Vf", "Hp", "Ac", "Ic", "Pc", "Ph" ))) %>% 
    arrange(site_specific) %>% 
    column_to_rownames(var = "site_specific") -> df_coryne_target_plot_heatmap
  
  row_order_site = c("Ra", "Mb", "Vf", "Hp", "Ac", "Ic", "Pc", "Ph")
  column_split_site = column_split
  colnames(df_coryne_target_plot_heatmap) = temp_name
  row_ha = rowAnnotation(number_of_samples= anno_barplot(rowSums(df_coryne_target_plot_heatmap>0) %>% unname(), ylim = c(0, 45)))
  col_fun = colorRamp2(c(0, 3, 5, 10, 20, 30, 90), c("#F5F5F5", "#F9D8A3", "#FDF1D1", "#E6F598", "#ABDDA4", "#66C2A5", "#3288BD"))
  # plot 3, 7
  p2 <- Heatmap(as.matrix(df_coryne_target_plot_heatmap),
                row_order = row_order_site,
                row_names_side = "left",
                show_row_dend = F, 
                show_column_dend = F,
                col = col_fun,
                column_gap = unit(1.5, "mm"),
                row_title = NULL,
                column_split = factor(column_split_site, levels = c("Pre-Tx", "Post-Tx-onIS≤3M", "Post-Tx-onIS-6M", "Post-Tx-onIS>12M", "Post-Tx-offIS-12M", "Post-Tx-offIS-24M", "HVs")),
                cluster_row_slices = TRUE,
                cluster_column_slices = FALSE, 
                column_title = NULL, 
                cluster_columns = FALSE, 
                cluster_rows = FALSE,
                right_annotation = row_ha,
                row_names_gp = gpar(fontsize = 6),
                column_names_gp = gpar(fontsize = 6),
                heatmap_legend_param = list(title = "Relative abundance (%)", at = c(0, 20, 40, 60, 80, 100)))
  return(p2)
}

p_C.tuberculostearicum_D <- p_coryne("C.tuberculostearicum_D")
p_C.kefirresidentii <- p_coryne("C.kefirresidentii")

p_C.tuberculostearicum_D
p_C.kefirresidentii
```

# Supplementary Figure 18
```{r}
df_supp18 <- read.delim2(file = "/Users/chey2/Desktop/DOCK8_paper_codes/Input_files/Supp_18_input.txt", header = TRUE, sep = "\t", stringsAsFactors = FALSE, check.names = FALSE)

df_supp18$Taxa <- factor(df_supp18$Taxa, levels = c("Other Staphylococcus", "Staphylococcus haemolyticus", "Staphylococcus capitis", "Staphylococcus hominis", "Staphylococcus epidermidis", "Staphylococcus aureus"))
df_supp18$subject_id <- factor(df_supp18$subject_id, levels = c("Pt01", "Pt02", "Pt03", "Pt04", "Pt05", "Pt06", "Pt07", "Pt08", "Pt09", "Pt10", "Pt11", "Pt12", "Pt13", "Pt14", "Pt15", "Pt16", "Pt17", "Pt18", "Pt19", "Pt20", "Pt21", "Pt22", "Pt23", "Pt24", "Hv01", "Hv02", "Hv03", "Hv04", "Hv05", "Hv06"))
df_supp18$Group <- factor(df_supp18$Group, levels = c("Pre-Tx", "Post-Tx-onIS-1M", "Post-Tx-onIS-2M", "Post-Tx-onIS-3M", "Post-Tx-onIS-6M", "Post-Tx-onIS>12M", "Post-Tx-offIS-12M", "Post-Tx-offIS-24M", "HVs"))
df_supp18$site_specific <- factor(df_supp18$site_specific, levels = c("Ra", "Mb", "Ac", "Ic", "Vf", "Pc", "Hp", "Ph"))
df_supp18$rela_abu <- as.numeric(df_supp18$rela_abu)

psupp18 <- df_supp18 %>% 
  ggplot(aes(x=as.factor(subject_id), y = rela_abu*100)) +
  geom_bar(stat = "identity", position = "fill", width = 0.8, aes(fill=Taxa)) + 
    # geom_bar(stat = "identity", position = "fill", width = 0.8) +
  scale_y_continuous(labels = function(x) x*100) +
  facet_grid(site_specific~Group, scales = "free_x", space = "free_x") +
  theme_bw() +
  theme(axis.text.x = element_text(angle = 90, size=7, hjust=1, vjust=0.3), axis.title.x=element_blank(), axis.text.y=element_text(size=7), axis.title.y=element_text(size=10)) +
  theme(panel.grid.minor=element_blank(), panel.grid.major=element_blank(), legend.title=element_text(size=10), legend.text=element_text(size=7)) +
  ylab("Mean relative abundance (%)") +
  scale_fill_manual("Classification", values =c("#DAD2C3","#6B8E23","#DAA520","#4682B4","#FD8D3C","#85002d"))

psupp18
```

# Supplementary Figure 19a
```{r}
# need results from figure3c
psupp19a <- df_supp14_paired_p %>% 
  ggplot(aes(x=Group, y = abu, fill = taxa)) +
  geom_bar(stat = "identity", position = "fill", width = 0.8) +
  scale_y_continuous(labels = function(x) x*100) +
  facet_grid(site_specific~subject_id,scales = "free_x",space = "free_x") +
  theme(axis.text.x = element_text(angle = 45, size=8, hjust=0.95,vjust=1), axis.title.x=element_blank(), axis.text.y=element_text(size=7), axis.title.y=element_text(size=12)) +
  theme(panel.grid.minor=element_blank(), panel.grid.major=element_blank(), legend.box = "vertical", legend.title=element_text(size=12), legend.text=element_text(size=8), legend.key.size=unit(0.5, "cm")) +
  theme(strip.text.x=element_text(size=8), strip.text.y=element_text(size=8)) +
  ylab("Mean relative abundance (%)") + 
  scale_fill_manual("Classification", values = color_pall_comb) +
  guides(fill=guide_legend(ncol=2))
psupp19a
```

# Supplementary Figure 19b
```{r}
# need results from figure3d
psupp19b <- df_3d %>% 
  filter(site_specific %in% c("Ra", "Vf", "Ac", "Ph")) %>% 
   ggplot(aes(x=new_grp, y = new_abu, fill = factor(final_taxa))) +
    geom_bar(stat = "identity", position = "fill", width = 0.8) +
    scale_y_continuous(labels = function(x) x*100) +
    facet_grid(site_specific~subject_id,scales = "free_x",space = "free_x") +
    theme(axis.text.x = element_text(angle = 45, size=8, hjust=0.95,vjust=1), axis.title.x=element_blank(), axis.text.y=element_text(size=7), axis.title.y=element_text(size=12)) +
   theme(panel.grid.minor=element_blank(), panel.grid.major=element_blank(), legend.box = "vertical", legend.title=element_text(size=12), legend.text=element_text(size=8), legend.key.size=unit(0.5, "cm")) + 
  theme(strip.text.x=element_text(size=8), strip.text.y=element_text(size=8)) +
  ylab("Mean relative abundance (%)") + 
    scale_fill_manual("Classification", values = color) +
  guides(fill=guide_legend(ncol=2))

psupp19b
```

# Supplementary Figure 19c
```{r}
# need results from supplementary figure 14b
df_supp14b %>% 
  filter(subject_id %in% c("Pt05", "Pt06", "Pt07", "Pt08", "Pt10", "Pt11")) %>% 
  filter(Group %in% c("Pre-Tx", "Post-Tx-offIS-12M", "Post-Tx-offIS-24M")) -> df_supp19c

df_supp19c$Group <- factor(df_supp19c$Group, levels = c("Pre-Tx", "Post-Tx-offIS-12M", "Post-Tx-offIS-24M"))
df_supp19c$subject_id <- factor(df_supp19c$subject_id, levels = c("Pt05", "Pt06", "Pt07", "Pt08", "Pt10", "Pt11"))

color_pall_comb_16s <- c(brewer.pal(5, 'Greens'),
               brewer.pal(5, 'Oranges')[c(1,3, 4, 5)],
               brewer.pal(11, 'BrBG')[4:3])

df_supp19c$site_specific <- factor(df_supp19c$site_specific, levels = c("Ra", "Mb", "Vf", "Hp", "Ac", "Ic", "Pc", "Ph"))

psupp19c <- df_supp19c %>% 
   ggplot(aes(x=as.factor(Group), y = rela_abu, fill = factor(final_taxa))) +
    geom_bar(stat = "identity", position = "fill", width = 0.8) +
    scale_y_continuous(labels = function(x) x*100) +
    facet_grid(site_specific~subject_id,scales = "free_x",space = "free_x") +
    theme(axis.text.x = element_text(angle = 45, size=8, hjust=0.95,vjust=1), axis.title.x=element_blank(), axis.text.y=element_text(size=7), axis.title.y=element_text(size=12)) +
   theme(panel.grid.minor=element_blank(), panel.grid.major=element_blank(), legend.box = "vertical", legend.title=element_text(size=12), legend.text=element_text(size=8), legend.key.size=unit(0.5, "cm")) + 
  theme(strip.text.x=element_text(size=8), strip.text.y=element_text(size=8)) +
  ylab("Mean relative abundance (%)") + 
    scale_fill_manual("Classification", values = color_pall_comb_16s) +
  guides(fill=guide_legend(ncol=2))

psupp19c
```

# Supplementary Figure 19d
```{r}
# need results from figure 3f
psupp19d <- df_supp19d %>% 
  filter(!Group=="Pre-Tx") %>% 
  filter(!site_specific %in% c("Mb", "Hp", "Ph")) %>% 
  ggplot(aes(x=as.factor(Group), y = relative_abu, fill = taxa)) +
  geom_bar(stat = "identity", position = "fill", width = 0.8) +
  scale_y_continuous(labels = function(x) x*100) +
  facet_grid(site_specific~subject_id,scales = "free_x",space = "free_x") +
  theme(axis.text.x = element_text(angle = 45, size=8, hjust=0.95,vjust=1), axis.title.x=element_blank(), axis.text.y=element_text(size=7), axis.title.y=element_text(size=12)) +
  theme(panel.grid.minor=element_blank(), panel.grid.major=element_blank(), legend.box = "vertical", legend.title=element_text(size=12), legend.text=element_text(size=8), legend.key.size=unit(0.5, "cm")) +
  theme(strip.text.x=element_text(size=8), strip.text.y=element_text(size=8)) +
  ylab("Mean relative abundance (%)") + 
  scale_fill_manual("Classification", values = color_pall_comb) +
  guides(fill=guide_legend(ncol=2))
psupp19d
```

# Supplementary Figure 20
```{r}
pre_postoff_12 <- c("Pt11", "Pt05", "Pt10", "Pt07", "Pt08", "Pt06", "Pt16", "Pt15", "Pt14", "Pt18", "Pt17", "Pt19", "Pt20")
pre_postoff_24 <-  c("Pt01", "Pt11", "Pt02", "Pt09", "Pt05", "Pt03", "Pt04", "Pt10", "Pt07", "Pt08", "Pt06", "Pt24")
pre_poston6 <-  c("Pt16", "Pt15", "Pt18", "Pt17", "Pt21", "Pt19", "Pt20")
pre_poston3 <-  c("Pt18", "Pt21", "Pt24", "Pt22", "Pt19", "Pt20", "Pt23")
pre_poston2 <- c("Pt22", "Pt20", "Pt23")
pre_poston1 <- c("Pt22", "Pt20", "Pt23")
pre_poston_12 <- c("Pt09", "Pt12", "Pt13")


df_supp20 <- read.delim("/Users/chey2/Desktop/DOCK8_paper_codes/Input_files/Supp_20_input.txt", header = TRUE, sep = "\t") # input

df_supp20$Group <- factor(df_supp20$Group, levels = c("Pre-Tx", "Post-Tx-onIS-1M", "Post-Tx-onIS-2M", "Post-Tx-onIS-3M", "Post-Tx-onIS-6M", "Post-Tx-offIS-12M", "Post-Tx-offIS-24M", "HVs", "Donor"))
df_supp20$site_specific <- factor(df_supp20$site_specific, levels = c("Ra", "Mb", "Hp", "Vf", "Ac", "Ic", "Pc", "Ph"))

Bray_distance_within_group <- function(in_put, category = c("Pre-Tx", "Post-Tx-onIS-3M", "Post-Tx-onIS-6M", "Post-Tx-offIS-12M", "Post-Tx-offIS-24M"), abu){
  # filter by sample 
  df1 <- df_supp20 %>% 
    filter(subject_id %in% {{in_put}}) %>% 
    filter(Group %in% {{category}})
  
  df1 %>% 
    group_by(variable, subject_id, site_specific, taxa, Group) %>% 
    summarise(mean_abu = mean({{abu}})) %>% 
    ungroup() -> df_2
  # remove unpaired sites

  if (length({{category}}) ==2){
    df_2 %>%
    select(subject_id, site_specific, Group) %>%
    unique() %>%
    group_by(subject_id, site_specific) %>%
    summarise(num=n()) %>% 
    filter(num == 1) %>%
    unite("target", subject_id:site_specific) %>% 
    ungroup() %>% 
    .$target -> need_remove
  }
  else if (length({{category}}) ==3){
    df_2 %>%
    select(subject_id, site_specific, Group) %>%
    unique() %>%
    group_by(subject_id, site_specific) %>%
    summarise(num=n()) %>% 
    filter(num <= 2) %>%
    unite("target", subject_id:site_specific) %>% 
    ungroup() %>% 
    .$target -> need_remove
  }
    
  df_2 %>% 
    filter(!paste(subject_id, site_specific, sep = "_") %in% need_remove) %>% 
    nest(data =c(variable, subject_id, taxa, mean_abu)) -> nested_df_2

    nested_df_2 %>% 
      mutate(distance = map(
        data, function(x){
          x <- x %>% select(-subject_id) %>% pivot_wider(names_from = taxa, values_from = mean_abu) %>% 
            column_to_rownames(var = "variable")
          dis <- vegdist(x, method="bray")
          return(dis)
        }
      )) -> nested_df_dist
    nested_df_dist %>% 
      mutate(withingrp_mean=unlist(lapply(distance, mean))) %>% 
      mutate(withingrp_sd = unlist(lapply(distance, sd))) %>% 
      select(site_specific, Group, withingrp_mean, withingrp_sd) -> nested_df_dist_mean_sd
    nested_df_dist %>% 
      select(site_specific, Group, distance) %>% 
      mutate(distance = purrr::map(distance, as.vector)) %>% 
      unnest(cols = distance) -> nested_df_dist_distance
    nested_df_dist$distance_new <- lapply(nested_df_dist$distance, as.numeric) 
    com_grp = nested_df_dist$Group %>% 
      unique() %>% 
      paste(collapse = " vs ")
    nested_df_dist %>% 
      filter(sapply(nested_df_dist$distance_new, length) > 0) %>% 
      group_by(site_specific) %>% 
      summarise(p_value=map_dbl(list(distance_new), ~wilcox.test(.x[[1]], .x[[2]])$p.value),
                                n1=length(distance_new[[1]]),
                                n2=length(distance_new[[2]]),
                                stat = map_dbl(list(distance_new), ~wilcox.test(.x[[1]], .x[[2]])$statistic)) %>% 
      ungroup() %>% 
      mutate(p_adj = p.adjust(p_value, method = "BH")) -> nested_df_dist_pvalue
    nested_df_dist_pvalue$grp = com_grp
    return(list(nested_df_dist_mean_sd, nested_df_dist_pvalue, nested_df_dist_distance))
}

Bray_postoff12 <- Bray_distance_within_group(pre_postoff_12, category = c("Pre-Tx", "Post-Tx-offIS-12M"), relative_abu)
Bray_postoff24 <- Bray_distance_within_group(pre_postoff_24, category = c("Pre-Tx", "Post-Tx-offIS-24M"), relative_abu)
Bray_poston6 <- Bray_distance_within_group(pre_poston6, category = c("Pre-Tx", "Post-Tx-onIS-6M"), relative_abu)
Bray_poston3 <- Bray_distance_within_group(pre_poston3, category = c("Pre-Tx", "Post-Tx-onIS-3M"), relative_abu)
Bray_poston2 <- Bray_distance_within_group(pre_poston2, category = c("Pre-Tx", "Post-Tx-onIS-2M"), relative_abu)
Bray_poston1 <- Bray_distance_within_group(pre_poston1, category = c("Pre-Tx", "Post-Tx-onIS-1M"), relative_abu)

dabest_prepare <- function(df, grp=c("Pre-Tx-1M", "Pre-Tx-2M", "Pre-Tx-3M", "Pre-Tx-6M", "Pre-Tx-12M", "Pre-Tx-24M")){
  df1 <- {{df}} %>% 
    mutate(grp=case_when(
       Group=="Pre-Tx" ~ {{grp}},
       TRUE ~ Group
    )) %>% 
    select(-Group)
  return(df1)
}

Bray_poston1_df <- dabest_prepare(Bray_poston1[[3]], grp="Pre-Tx-1M") 
Bray_poston2_df <- dabest_prepare(Bray_poston2[[3]], grp="Pre-Tx-2M")
Bray_poston3_df <- dabest_prepare(Bray_poston3[[3]], grp="Pre-Tx-3M")
Bray_poston6_df <- dabest_prepare(Bray_poston6[[3]], grp="Pre-Tx-6M")
Bray_postoff12_df <- dabest_prepare(Bray_postoff12[[3]], grp="Pre-Tx-12M")
Bray_postoff24_df <- dabest_prepare(Bray_postoff24[[3]], grp="Pre-Tx-24M")

Bray_all_d <- rbind(Bray_poston1_df, Bray_poston2_df, Bray_poston3_df, Bray_poston6_df, Bray_postoff12_df, Bray_postoff24_df)

dabest_plot_fun <- function(site){
  Bray_all_d %>% 
    filter(site_specific=={{site}}) -> Bray_all_temp
  Bray_all_temp$site_specific <- droplevels(Bray_all_temp$site_specific)
  multi_groups <- load(Bray_all_temp,
                     x = grp, y = distance,
                     idx = list(
                       c("Pre-Tx-1M", "Post-Tx-onIS-1M"),
                       c("Pre-Tx-2M", "Post-Tx-onIS-2M"),
                       c("Pre-Tx-3M", "Post-Tx-onIS-3M"),
                       c("Pre-Tx-6M", "Post-Tx-onIS-6M"),
                       c("Pre-Tx-12M", "Post-Tx-offIS-12M"),
                       c("Pre-Tx-24M", "Post-Tx-offIS-24M")))
  p <- multi_groups %>%
  mean_diff() %>%
  dabest_plot(show_zero_dot=FALSE, raw_marker_size=0.5, tufte_size=0.3, es_marker_size=0.5, es_line_size=0.3, raw_marker_spread=1.5, swarm_x_text=6, swarm_y_text=6, contrast_x_text=6, contrast_y_text=6, raw_marker_alpha=0.5, swarm_label="Bray Curtis distance", contrast_label="Mean difference", custom_palette="igv")
  results_temp <- multi_groups %>%
  mean_diff() %>% .$permtest_pvals %>% .$pval_for_tests 
  do.call(rbind, results_temp) -> results
  colnames(results) = "pvalue"
  results <- as.data.frame(results)
  results$site <- {{site}}
  results$p_adj <- p.adjust(results$pvalue, method = "BH")
  return(list(p, results))
}

site_lt <- c("Ra", "Mb", "Vf", "Ac", "Ic", "Pc", "Ph")
psupp20_7_list <- map(site_lt, dabest_plot_fun)

# Hp site
Bray_all_d %>% 
  filter(site_specific=="Hp") %>% 
  filter(!grp %in% c("Pre-Tx-2M", "Post-Tx-onIS-2M")) -> Bray_all_Hp
  
Bray_all_Hp$site_specific <- droplevels(Bray_all_Hp$site_specific)

Hp_groups <- load(Bray_all_Hp,
                     x = grp, y = distance,
                     idx = list(
                       c("Pre-Tx-3M", "Post-Tx-onIS-3M"),
                       c("Pre-Tx-6M", "Post-Tx-onIS-6M"),
                       c("Pre-Tx-12M", "Post-Tx-offIS-12M"),
                       c("Pre-Tx-24M", "Post-Tx-offIS-24M")))

psupp20_1 <- Hp_groups %>%
  mean_diff() %>%
  dabest_plot(show_zero_dot=FALSE, raw_marker_size=0.5, tufte_size=0.3, es_marker_size=0.5, es_line_size=0.3, raw_marker_spread=1.5, swarm_x_text=6, swarm_y_text=6, contrast_x_text=6, contrast_y_text=6, raw_marker_alpha=0.5, swarm_label="Bray Curtis distance", contrast_label="Mean difference", custom_palette="igv")

psupp20_7 <- cowplot::plot_grid(plotlist = psupp20_7_list %>% lapply(., "[[", 1), ncol = 3)
psupp20_1
```

# Supplementary Figure 21
```{r}
# need results from figure3g
asso_all_plots <- map(all_sites, ass_plot)
psupp21 <- cowplot::plot_grid(plotlist = asso_all_plots)
psupp21
```

# Supplementary Figure 22a
```{r}
# need results from figure4a
psupp22 <- df_4a %>%  
  ggplot(aes(x=Group, y=log_num)) + 
  geom_boxplot(lwd = 0.15, outlier.size = 0.1, outlier.shape = NA, width=0.4) +
  geom_dotplot(binaxis='y', stackdir='center',stackratio=1.8, dotsize=1, aes(fill=grp), color=NA) +
  # geom_smooth(method = "loess", se = FALSE, aes(group = KEY), span = 0.7, fill="lightgrey", size=0.3) + 
  facet_grid(KEY~., scales = "free") + 
  scale_fill_manual("Group", values = c("#dfd049", "#1E6CB5", "#936bc0")) + 
  theme(axis.text.x = element_text(angle = 45, size=8, hjust=0.95,vjust=1), axis.title.x=element_blank(), axis.text.y=element_text(size=8), axis.title.y=element_blank()) +
  theme(panel.grid.minor=element_blank(), panel.grid.major=element_blank(), legend.title=element_text(size=10), legend.text=element_text(size=7)) + scale_y_continuous(
  labels = scales::number_format(accuracy = 0.1)) 
psupp22
```

# Supplementary Figure 22b
```{r}
df_supp22 <- read.delim("/Users/chey2/Desktop/DOCK8_paper_codes/Input_files/Supp_22b_input.txt", header = TRUE, sep = "\t") # input

df_supp22$grp <- factor(df_supp22$grp, levels = c("Pre-Tx", "Post-Tx-onIS<=3M", "Post-Tx-onIS-6M", "Post-Tx-onIS>=12M", "Post-Tx-offIS-12M", "Post-Tx-offIS-24M"))

spearman_viral_load <- function(immune){
  viral_load <- df_supp22[["log_total_load"]]
  immune_var <- df_supp22[[{{immune}}]]
  spearman_result <- cor.test(viral_load, immune_var, method = "spearman") %>% tidy()
  spearman_result$Immune <- {{immune}}
  return(spearman_result)
}

immune_list <- c("CD19", "CD4", "CD8", "NK")

spearman_viral_load_temp <- map(immune_list, spearman_viral_load)

spearman_viral_load_out <- spearman_viral_load_temp %>% 
  do.call(rbind, .) %>% 
  mutate(p.adj = round(p.adjust(p.value, method = "BH"), 2))

spearman_output_plot <- function(variable){
  # variable_str <- deparse(substitute(variable))
  R <- spearman_viral_load_out %>% filter(Immune==variable) %>% .$estimate %>% unname()
  p.adj <- spearman_viral_load_out %>% filter(Immune==variable) %>% .$p.adj %>% unname()
  spearman_sta <- paste0("R=", round(R,2), ", p=", p.adj)

  p1 <- df_supp22 %>% 
    ggplot(aes(x=log_total_load, y=eval(parse(text = variable)))) +
    geom_point(aes(color=grp), size=0.4) +
    geom_smooth(method = "lm", color = "black", size=0.5) +
    scale_color_manual(values = c("#C7E9B4", "#47af6f","#7FCDBB", "#FEE090", "#1D91C0", "#081D58")) +
    theme_bw() +
    theme(axis.title.x=element_blank()) +
    theme(legend.position = "none") +
    ggtitle(spearman_sta) +
    theme(plot.title = element_text(hjust = 0.05, vjust = -8.5, size = 6)) +
    theme(panel.grid.minor=element_blank(), panel.grid.major=element_blank()) +
    ylab({{variable}})

  return(p1)
}

spearman_total_load_plot_list <- map(immune_list, spearman_output_plot)
psupp22b <- cowplot::plot_grid(plotlist = spearman_total_load_plot_list, ncol = 4)
```

# Supplementary Figure 23a
```{r}
# need results from figure4b
nano.obj.clean.new.sct.default %>% 
  arrange(p_val_adj) %>% 
  arrange(-avg_log2FC) %>% 
  group_by(cluster) %>%
  slice_head(n = 10) %>%
  ungroup() -> top10

nano.obj.clean.new.sct@active.ident <- factor(nano.obj.clean.new.sct@active.ident, levels = c("KC_basal", "KC_suprabasal_1", "KC_suprabasal_2", "KC_Granular", "Immune cells", "Fibroblasts"))

psupp23a <- DoHeatmap(nano.obj.clean.new.sct, features = top10$gene, group.colors = c("#f54888",  "#009166", "#9be68c", "#006bbf", "#7a65dd", "#e5b930"), size = 3) + theme(text = element_text(size = 4)) + scale_fill_gradient2(low = rev(c('#d1e5f0','#67a9cf','#2166ac')), mid = "white", high = rev(c('#b2182b','#ef8a62','#fddbc7')), midpoint = 0) + guides(color="none") + theme(text=element_text(size=5)) 
psupp23a

```

# Supplementary Figure 23b
```{r}
# need results from figure4b
p_KRT16 <- ImageFeaturePlot(nano.obj.clean.new.sct, features = "KRT16", fov = "EHKSlide1clean", border.color = "NA")
p_KRT6ABC <- ImageFeaturePlot(nano.obj.clean.new.sct, features = "KRT6ABC", fov = "EHKSlide1clean", border.color = "NA")
p_krtMHCI <- ImageFeaturePlot(nano.obj.clean.new.sct, features = "MHCI", fov = "EHKSlide1clean", border.color = "NA")
p_krtSTAT1 <- ImageFeaturePlot(nano.obj.clean.new.sct, features = "STAT1", fov = "EHKSlide1clean", border.color = "NA")
p_krtHLADRA <- ImageFeaturePlot(nano.obj.clean.new.sct, features = "HLA-DRA", fov = "EHKSlide1clean", border.color = "NA")
p_krtCXCL10 <- ImageFeaturePlot(nano.obj.clean.new.sct, features = "CXCL10", fov = "EHKSlide1clean", border.color = "NA")
p_krtCXCL9 <- ImageFeaturePlot(nano.obj.clean.new.sct, features = "CXCL9", fov = "EHKSlide1clean", border.color = "NA")

P_feature <- FeaturePlot(nano.obj.clean.new.sct, features = c("KRT16", "KRT6ABC", "MHCI", "STAT1", "HLA-DRA", "CXCL10", "CXCL9") , split.by = "group", pt.size = 0.05)
```

# Supplementary Figure 23c
```{r}
# need results from figure4b
gene_list <- c("KRT14", "KRT16", "KRT6A/B/C")
psupp23c <- vplot_gene_cluster_obj(nano.obj.all.KC.sct, gene_list,c("KC_basal", "KC_suprabasal_1", "KC_suprabasal_2", "KC_Granular"))
psupp23c
```

